src/modules/control_allocator/ActuatorEffectiveness/ActuatorEffectivenessControlSurfaces.cpp

```diff
@@ -151,32 +164,25 @@ bool ActuatorEffectivenessControlSurfaces::addActuators(Configuration &configura
 	return true;
 }
 
-void ActuatorEffectivenessControlSurfaces::applyFlapsAirbrakesWheel(float flaps_control, float airbrakes_control,
-		float wheel_control, int first_actuator_idx,
-		ActuatorVector &actuator_sp) const
+void ActuatorEffectivenessControlSurfaces::applyFlaps(float flaps_control, int first_actuator_idx, float dt,
+		ActuatorVector &actuator_sp)
 {
-	for (int i = 0; i < _count; ++i) {
-		switch (_params[i].type) {
-		// TODO: check sign
-		case ActuatorEffectivenessControlSurfaces::Type::LeftFlaps:
-			actuator_sp(i + first_actuator_idx) += flaps_control;
-			break;
-
-		case ActuatorEffectivenessControlSurfaces::Type::RightFlaps:
-			actuator_sp(i + first_actuator_idx) -= flaps_control;
-			break;
+	_flaps_setpoint_with_slewrate.update(flaps_control, dt);
 
-		case ActuatorEffectivenessControlSurfaces::Type::Airbrakes:
-			actuator_sp(i + first_actuator_idx) += airbrakes_control;
-			break;
+	for (int i = 0; i < _count; ++i) {
+		// map [0, 1] to [-1, 1]
+		// TODO: this currently only works for dedicated flaps, not flaperons
```

Comment:
Correct me if I'm wrong, but currently to get a control surfaces to move to PWM_MIN you have to set it's actuator_sp to -1. There is no CENTER setting that would allow you to say "map actuator_sp 0 to XX PWM". 
The flaps_setpoint is a signed desired configuration (0 deployment to full deployment), which means I have to map [0,1] to [-1, 1] to achieve full range on dedicated flaps. When I though want to allocate the flaps configuration to ailerons, which can move up and down, the [0,1] could be directly applied.

Reply:
Right, we probably don't get around to make some kind of distinction for these 2 use-cases.

Reply:
Yes we can discuss that in a follow up.